snuba/web/rpc/v1/endpoint_get_traces.py

```diff
@@ -0,0 +1,289 @@
+    TraceColumn.Name.NAME_TRACE_ID: {
+        AttributeKey.Type.TYPE_STRING,
+    },
+    TraceColumn.Name.NAME_START_TIMESTAMP: {
+        AttributeKey.Type.TYPE_STRING,
+        AttributeKey.Type.TYPE_INT,
+        AttributeKey.Type.TYPE_FLOAT,
+    },
+}
+
+
+def _column_to_expression(trace_column: TraceColumn, conditions=None) -> Expression:
+    if trace_column.name == TraceColumn.Name.NAME_TOTAL_SPAN_COUNT:
+        return f.count(
+            alias=_COLUMN_TO_NAME[trace_column.name],
+        )
+    if trace_column.name == TraceColumn.Name.NAME_FILTERED_SPAN_COUNT:
+        return f.countIf(
+            conditions,
+            alias=_COLUMN_TO_NAME[trace_column.name],
+        )
+    if trace_column.name == TraceColumn.Name.NAME_START_TIMESTAMP:
+        return f.CAST(
+            f.min(column("start_timestamp")),
+            _TYPES_TO_CLICKHOUSE[trace_column.type],
+            alias=_COLUMN_TO_NAME[trace_column.name],
+        )
+    if trace_column.name == TraceColumn.Name.NAME_ROOT_SPAN_NAME:
+        return f.anyIf(
+            column("name"),
+            f.equals(column("is_segment"), True),
+            alias=_COLUMN_TO_NAME[trace_column.name],
+        )
+    if (
+        trace_column.name in _COLUMN_TO_NAME
+        and trace_column.type in _POSSIBLE_TYPES.get(trace_column.name, {})
+    ):
+        return f.CAST(
+            column(
+                _COLUMN_TO_NAME[trace_column.name],
+            ),
+            _TYPES_TO_CLICKHOUSE[trace_column.type],
+            alias=_COLUMN_TO_NAME[trace_column.name],
+        )
+    raise BadSnubaRPCRequestException(
+        f"{trace_column.name} had an unknown or unset type: {trace_column.type}"
+    )
+
+
+def _convert_order_by(
+    order_by: Sequence[GetTracesRequest.OrderBy],
+) -> Sequence[OrderBy]:
+    res: list[OrderBy] = []
+    for x in order_by:
+        direction = OrderByDirection.DESC if x.descending else OrderByDirection.ASC
+        res.append(
+            OrderBy(
+                direction=direction,
+                expression=_column_to_expression(x.column),
+            )
+        )
+    return res
+
+
+def _build_query(request: GetTracesRequest) -> Query:
+    trace_item_filter_expressions = trace_item_filters_to_expression(request.filter)
+    selected_columns = []
+
+    for trace_column in request.columns:
+        expression = _column_to_expression(trace_column, trace_item_filter_expressions)
+        selected_columns.append(
+            SelectedExpression(
+                name=_COLUMN_TO_NAME[trace_column.name],
+                expression=expression,
+            )
+        )
+
+    entity = Entity(
+        key=EntityKey("eap_spans"),
+        schema=get_entity(EntityKey("eap_spans")).get_data_model(),
+        sample=None,
+    )
+    res = Query(
+        from_clause=entity,
+        selected_columns=selected_columns,
+        condition=base_conditions_and(
+            request.meta,
+        ),
```

Comment:
This isn't applying the user conditions. Wouldn't that result in a full scan of the table across all orgs? Or am I missing something?

Reply:
Yes, this is a mistake.